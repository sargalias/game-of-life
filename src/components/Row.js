import React from 'react';
import Cell from './Cell';

const Row = (props) => (
  <div
    className="row"
  >
    {props.rowData.map((cell, index) => (
      <Cell
        key={index}
        rowId={props.rowId}
        colId={index}
        value={cell}
        bgColor={props.rowColorData && props.rowColorData[index]}
        onCellClick={props.onCellClick}
      />
    ))}
  </div>
);

export default Row;
