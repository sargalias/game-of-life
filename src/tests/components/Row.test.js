import React from 'react';
import { shallow } from 'enzyme';
import Row from '../../components/Row';

test('row should render correctly', () => {
  const foo = () => {};
  const rowData = [0, 0, 1, 1];
  const wrapper = shallow(<Row rowData={rowData} rowId={5} onCellClick={foo} />);
  expect(wrapper).toMatchSnapshot();
});
