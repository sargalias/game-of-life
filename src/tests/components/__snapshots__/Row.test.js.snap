// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`row should render correctly 1`] = `
<div
  className="row"
>
  <Cell
    colId={0}
    key="0"
    onCellClick={[Function]}
    rowId={5}
    value={0}
  />
  <Cell
    colId={1}
    key="1"
    onCellClick={[Function]}
    rowId={5}
    value={0}
  />
  <Cell
    colId={2}
    key="2"
    onCellClick={[Function]}
    rowId={5}
    value={1}
  />
  <Cell
    colId={3}
    key="3"
    onCellClick={[Function]}
    rowId={5}
    value={1}
  />
</div>
`;
